### ISSUE TITLE
TableView consumes all click events.

### ISSUE THREAD

**Issue by omar753sahl** [evrencoskun/TableView#147]
I have an EditText as a table cell, But it doesn't get any click events so I can't edit the text.
Is there any way to do it?

Thank you.

**Comment by orlytops** [comment]
I have same problem any solutions for this?

**Comment by evrencoskun** [comment]
Hi guys, 

To be able to edit any cell, you need to use EditText instead of TextView. However, using EditText is very [expensive](https://issuetracker.google.com/issues/37015285).

So, why you have this problem? You probably have a focus problem.

**Comment by orlytops** [comment]
same with button when I click a button inside the cell  the click listener for the button doesn't trigger.

**Comment by orlytops** [comment]
I think it each cell doesn't support click events for its children

**Comment by evrencoskun** [comment]
@orlytops That is totally wrong. check this [comment ](https://github.com/evrencoskun/TableView/issues/52#issuecomment-368298186)

**Comment by orlytops** [comment]
Yeah I think the listener that's been triggered there is the cell Click listener not the button Click listener inside the Cell... I can trigger the button when I double click the cell but I need to click it once. @evrencoskun

**Comment by orlytops** [comment]
I'm declareing the button listener inside the `onBindCellViewHolder`

here's the sample code

```
@Override
  public void onBindCellViewHolder(AbstractViewHolder holder, Object cellItemModel, int columnPosition, int rowPosition) {

    CellViewHolder CellViewHolder = (CellViewHolder) holder;
      controlsCellViewHolder.sendButton.setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View v) {
          Log.i("Clicked", "Button Cliked");
        }
      });

  }
```

**Comment by evrencoskun** [comment]
@orlytops ye the reason that you have to click double times is the component is not focused. At the first click, it gets focus. 

@omar753sahl  
I have just changed TextView with EditText. It's the result.  

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/4228262/43888762-25fd5dec-9bcb-11e8-9f7d-7a47af86a23e.gif)

BTW, It's not about the TableView actually, It's about `https://stackoverflow.com/questions/38242530/recyclerview-single-click-not-working-but-works-when-click-twice`.

**Comment by orlytops** [comment]
@evrencoskun but I need it once do you have any work around about this?

**Comment by omar753sahl** [comment]
@orlytops Here's a workaround for this.
I changed this method in `AbstractItemClickListener` from this:
``` java
 @Override
    public boolean onInterceptTouchEvent(RecyclerView view, MotionEvent e) {
        return mGestureDetector.onTouchEvent(e);
    }
```
to this:
``` java
 @Override
    public boolean onInterceptTouchEvent(RecyclerView view, MotionEvent e) {
        mGestureDetector.onTouchEvent(e);
        return false;
    }
```

**Comment by brendajerop** [comment]
I was able to edit the editText when it's clicked once by requesting focus and showing the soft keyboard in TableViewListener.

       `@Override
        public void onCellClicked(@NonNull final RecyclerView.ViewHolder    cellView, final int column, final int row) {
        cellText = (EditText)cellView.itemView.findViewById(R.id.cell_data);
        cellText.requestFocus();
        //editTexts are initially disabled to improve on scrolling
        cellText.setEnabled(true);
        //show keyboard
        InputMethodManager imm = (InputMethodManager)       mContext.getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.showSoftInput(cellText, InputMethodManager.SHOW_IMPLICIT);`

![ezgif com-gif-maker](https://user-images.githubusercontent.com/20903830/44502236-3c7f8f80-a699-11e8-9022-8fa6114cbdff.gif)

**Comment by somodikrisztian1** [comment]
@omar753sahl Your solution is perfect! This should be more straightforward in this otherwise fantastic library.

**Comment by sgallego** [comment]
See #253

**Comment by Jhon354202** [comment]
> @ omar753sahl您的解决方案是完美的！在这个原本很棒的库中，这应该更直接。

I have no effect with this configuration. What's the reason I'm missing?
